proxy/src/outbound.rs

```diff
@@ -51,81 +51,78 @@ impl<B> Outbound<B> {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub enum Destination {
-    LocalSvc(FullyQualifiedAuthority),
-    External(SocketAddr),
-}
-
-impl From<FullyQualifiedAuthority> for Destination {
-    #[inline]
-    fn from(authority: FullyQualifiedAuthority) -> Self {
-        Destination::LocalSvc(authority)
-    }
+pub enum Dst {
```

Comment:
This is minor and I could be totally out of line, but personally i'm not sure if an `enum` where each variant owns a `Protocol` is any clearer than a tuple of (`enum`, `Protocol`)...

Reply:
The intent is that a `Dst` is meaningless without a protocol.  It doesn't make sense (to me) for `Dst` _not_ to have a protocol.

Reply:
That makes sense. I guess my instinct is to not duplicate fields across enum variants when they're the same?

Reply:
I agree that Dst is meaningless without a protocol. However, the current structure actually implies that we might add new variants that don't have a protocol since whether there's a protocol field is determined per-variant. If every Dst has a protocol then it should be like @hawkw suggested:
```rust
pub struct Dst {
    protocol: Protocol,
    binding: DstBinding,
}
enum DstBnding {
    Unbound(FullyQualifiedAuthority),
    Bound(SocketAddr),
}
```

(Just like `bind::BindProtocol`.)